File: go/array/1732. 找到最高海拔.go
package array

import "math"

func largestAltitude(gain []int) int {
	res := make([]int, len(gain)+1)
	for i := 0; i < len(gain); i++ {
		res[i+1] = res[i] + gain[i]
	}
	max := math.MinInt32
	for _, v := range res {
		if v > max {
			max = v
		}
	}
	return max
}

File: python/数组/1732. 找到最高海拔.py
from typing import List
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        res = [0 for i in range(len(gain)+1)]
        for i in range(len(gain)):
            res[i+1] = res[i] + gain[i]
        return max(res)
s = Solution()
gain = [-4,-3,-2,-1,4,3,2]
print(s.largestAltitude(gain))
